controllers/root.js

```diff
@@ -42,13 +44,19 @@ var Root = (function() {
     		IDM.check_token(auth_token, function (user_info) {
 
                 if (config.azf.enabled) {
-                    
-                    AZF.check_permissions(auth_token, user_info, req, function () {
-
-                        redir_request(req, res, user_info);
 
+                    AZF.check_permissions(auth_token, user_info, req, function () {
+                        if (config.work_mode === 'validate'){
+                            redir_request_validate(req, res, user_info);
+                        } else {
+                            redir_request(req, res, user_info);
+                        }
                     }, function (status, e) {
-                        if (status === 401) {
+
+                        if (status === 401 && config.work_mode === 'validate') {
+                            log.error('User access-token not authorized: ', e);
+                            res.status(401).send('User access-token not authorized');
+                        } else if (status === 401 ) {
```

Comment:
I don't see the difference between these two cases. The response in both cases is 401. Why do you duplicate the code?